engine/src/main/java/org/terasology/world/block/BlockLifecycleEvent.java

```diff
@@ -32,11 +36,46 @@ public BlockLifecycleEvent(TIntList positions, BlockEntityRegistry registry) {
         this.positions = positions;
     }
 
-    public Iterable<Vector3i> getBlockPositions() {
-        return () -> new BlockPositionIterator(positions, registry);
+    @NotNull
+    @Override
+    public Iterator<Vector3ic> iterator() {
```

Comment:
What exactly does this iterator do?

The `TIntList positions` is flattened list of positions, right? So if this event affects
```
[(0,0,0), (1,2,3), (4,5,6)]
```
then `positions` would be:
```
[0,0,0, 1,2,3, 4,5,6]
```

But whether a block listed in `positions` is actually returned depends on whether there's a _permanent block entity_ registered or not? 
Doesn't this lead to a mismatch between the size as returned by `blockCount` and the actual amount of positions returned by the iterator?